letsencrypt-win-simple/Scripts/ImportExchange.ps1

```diff
@@ -1,56 +1,95 @@
+        
+    }
 
+    # Make sure variable is defined
+    $null = Get-ChildItem $CertInStore.PSPath -ErrorAction Stop
+    # Cert must exist in the personal store of machine to bind to exchange services
+    if($CertInStore.PSPath -notlike "*LocalMachine\My\*"){
+        $SourceStoreScope = 'LocalMachine'
+        $SourceStorename = $CertInStore.PSParentPath.split("\")[-1]
 
+        $SourceStore = New-Object  -TypeName System.Security.Cryptography.X509Certificates.X509Store  -ArgumentList $SourceStorename, $SourceStoreScope
+        $SourceStore.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadOnly)
 
-            $DestStoreScope = 'LocalMachine'
-            $DestStoreName = 'My'
+        $cert = $SourceStore.Certificates | Where-Object {$_.thumbprint -eq $CertInStore.Thumbprint}
 
-            $DestStore = New-Object  -TypeName System.Security.Cryptography.X509Certificates.X509Store  -ArgumentList $DestStoreName, $DestStoreScope
-            $DestStore.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
-            $DestStore.Add($cert)
 
 
-            $SourceStore.Close()
-            $DestStore.Close()
+        $DestStoreScope = 'LocalMachine'
+        $DestStoreName = 'My'
 
-            $CertInStore = Get-ChildItem -Path Cert:\LocalMachine\My -Recurse | Where-Object {$_.thumbprint -eq $NewCertThumbprint} | Sort-Object -Descending | Select-Object -f 1
-        }
-        Enable-ExchangeCertificate -Services $ExchangeServices -Thumbprint $CertInStore.Thumbprint -ErrorAction Stop
-        "Cert thumbprint set to the following exchange services: $ExchangeServices"
-    }catch{
-        "Cert thumbprint was not set successfully"
-        "Error: $($Error[0])"
+        $DestStore = New-Object  -TypeName System.Security.Cryptography.X509Certificates.X509Store  -ArgumentList $DestStoreName, $DestStoreScope
+        $DestStore.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
+        $DestStore.Add($cert)
+
+
+        $SourceStore.Close()
+        $DestStore.Close()
+
+        $CertInStore = Get-ChildItem -Path Cert:\LocalMachine\My -Recurse | Where-Object {$_.thumbprint -eq $NewCertThumbprint} | Sort-Object -Descending | Select-Object -f 1
+    }
+    Enable-ExchangeCertificate -Services $ExchangeServices -Thumbprint $CertInStore.Thumbprint -Force -ErrorAction Stop
+    "Cert thumbprint set to the following exchange services: $ExchangeServices"
+
+    
+    if(-not $LeaveOldExchangeCerts){
+        "Old Exchange certificates being cleaned up"
+        Get-ExchangeCertificate -DomainName $CertInStore.Subject.split("=")[1] | Where-Object -FilterScript {
+            $_.Thumbprint -ne $newCertThumbprint
+        } | Remove-ExchangeCertificate -Confirm:$false
     }
-}else{
-    "Cert thumbprint not found in the cert store... which is strange because it should be there."
+
+}catch{
+    "Cert thumbprint was not set successfully"
+    "Error: $($Error[0])"
```

Comment:
> "Error: $($Error[0])"

Could we use `throw $_` instead, to throw a real error? Or after this row maybe?

Reply:
That would be a question for @WouterTinus . I don't see the difference really. Either way it'll be logged.

Disclaimer: I'm not a proper developer, so I'm not really well-versed error handling. Outputting them always works fine enough for my needs.